[expo-screen-orientation] Update documentation (#6838)
expo/expo: docs/pages/versions/unversioned/sdk/screen-orientation.md

This API is pre-installed in [managed](../../introduction/managed-vs-bare/#managed-workflow) apps. It is not yet available for [bare](../../introduction/managed-vs-bare/#bare-workflow) React Native apps.

### Warning

> This module needs additional configuration to work properly on iPads. Unfortunately, it will disable support for the split view.

Reviewer: Could you elaborate a bit on what does it mean to _work properly on iPads_, why doesn't this module _work properly_ by default, what are the advantages and disadvantages of adding `requireFullScreen: true` property?

#### Managed workflow

> Open your app.json and add the following inside of the "expo" field:

Reviewer: ```suggestion
Open your `app.json` and add the following inside of the `"expo"` field:
```

```json
"ios": {
    "requireFullScreen": true,
}
```

#### Bare workflow

> Tick the `Requires Full Screen` checkbox in XCode. It should be located under `Project Target > General > Deployment Info`.

Reviewer: ```suggestion
Tick the `Requires Full Screen` checkbox in Xcode. It should be located under `Project Target > General > Deployment Info`.
```

> The Apple added support for the `split view` mode to iPads in iOS 9. This changed how the screen orientation is handled by the system. Briefly, for the iOS, your iPad is always in the landscape mode unless you open two applicationes side by side. To use the `ScreenOrientation` module you need to disable support for this feature. For more information about the `split view` mode, check out [the official Apple documentation](https://support.apple.com/en-us/HT207582).

Reviewer: ```suggestion
Apple added support for _split view_ mode to iPads in iOS 9. This changed how the screen orientation is handled by the system. To put the matter shortly, for the iOS, your iPad is always in the landscape mode unless you open two applications side by side. In order to be able to lock screen orientation using this module you will need to disable support for this feature. For more information about the _split view_ mode, check out [the official Apple documentation](https://support.apple.com/en-us/HT207582).
```

> Open your `app.json` and add the following inside of the `"expo"` field:

Reviewer: How about putting the `expo` to inside the json? so it looks kinda like:

```
{
  expo: {
    ...
    ios: {
      ...
      requireFullScreen: true
```

#### Error Codes

> - `ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK` - an invalid [`OrientationLock`](#screenorientationorientationlock) was passed in (**isOnly**).

Reviewer: ```suggestion
- `ERR_SCREEN_ORIENTATION_INVALID_ORIENTATION_LOCK` - an invalid [`OrientationLock`](#screenorientationorientationlock) was passed in (**iOS only**).
```

I guess

- **`OrientationLock.OTHER`** -- A platform specific orientation. This is not a valid policy that can be applied in [`lockAsync`](#screenorientationlockasyncorientationlock).
- **`OrientationLock.UNKNOWN`** -- An unknown screen orientation lock. This is not a valid policy that can be applied in [`lockAsync`](#screenorientationlockasyncorientationlock).

> > **Note** `OrientationLock.ALL` and `OrientationLock.PORTRAIT` are invalid on devices which not support `OrientationLock.PORTRAIT_DOWN`.

Reviewer: ```suggestion
> **Note** `OrientationLock.ALL` and `OrientationLock.PORTRAIT` are invalid on devices which do not support `OrientationLock.PORTRAIT_DOWN`.
```
